lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1989,6 +1990,20 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
   {
     LLDB_SCOPED_TIMERF("%s (getStdlibModule)", m_description.c_str());
     const bool can_create = true;
+
+    // Report progress on module importing by using a callback function in swift::ASTContext
+    Progress progress("Importing Swift standard library modules");
+    swift_ast_sp->m_ast_context_ap->SetPreModuleImportCallback(
+      [&progress](llvm::StringRef module_name, bool is_overlay) {
+        progress.Increment(1, (is_overlay ? module_name.str() + " (overlay)" : module_name.str()));
+      }
+);
+
+    // Clear the callback function on scope exit to prevent an out-of-scope access of the progress local variable
+    auto on_exit = llvm::make_scope_exit([&](){
+      swift_ast_sp->m_ast_context_ap->SetPreModuleImportCallback([](llvm::StringRef module_name, bool is_overlay) {});
```

Comment:
If I understand correctly, today we'll always emit a generic progress of "Importing [overlay] module X" whenever a module is imported. With this change, we'll have better progress reporting messages, but won't report anything else that happens to be imported outside of these functions. Perhaps re-introduce the generic progress message in case modules are imported outside these functions?

Reply:
Good point, I hadn't thought about the scenario of module imports being triggered in LLDB outside of these functions. If that does happen and the callback function is generic then LLDB might not be able to give a detailed progress report (since the generic callback wouldn't have the context of which modules were imported or where the import took place from) but it still indicates that LLDB caused a Swift module import which the user would want to know.